### ISSUE TITLE
QgsProject doesn't contain relations

### ISSUE THREAD

**Issue by Gustry** [3liz/qgis-atlasprint#36]
In the plugin, we have an issue with missing relation in QgsProject.

For example, I have an atlas on French department. And I have a relation to a French cities layer. In my atlas, I'm display a few cities in a table belonging to the current department.

In QGIS Desktop, it's working well, but using the atlasprint plugin, the relation table is empty.

```python
    relations = project.relationManager()
    QgsMessageLog.logMessage('RELATIONS', 'atlasprint', Qgis.Info)
    for relation in relations.relations().keys():
        QgsMessageLog.logMessage(relation, 'atlasprint', Qgis.Info)
    QgsMessageLog.logMessage('FIN des relations', 'atlasprint', Qgis.Info)
```

Nothing in my logs. I added this code in the `core.py` for instance.

The problem is we have no access when the QgsProject is created, the object comes from `executeRequest` method in `QgsService` https://github.com/3liz/qgis-atlasprint/blob/master/atlasprint/service.py#L108 which is in QGIS core.

So it's maybe an upstream bug?

This is a regression from 1.2.4

CC @rldhont @dmarteau

**Comment by Gustry** [comment]
For a workaround with  @rldhont , we removed the use of the relation in the atlas. We used a plain QGIS expression to filter my attribute table:

```
 "insee_dep" =  attribute(  @atlas_feature , 'insee_dep')
```

And it's working.

**Comment by rldhont** [comment]
Relations are read in the QGIS Project document after `readProjectWithContext` emission by `QgsProject`.
* read in server config cache https://github.com/qgis/QGIS/blob/master/src/server/qgsconfigcache.cpp#L47
* read https://github.com/qgis/QGIS/blob/master/src/core/qgsproject.cpp#L1075 https://github.com/qgis/QGIS/blob/master/src/core/qgsproject.cpp#L1081 https://github.com/qgis/QGIS/blob/master/src/core/qgsproject.cpp#L1120
* readProjectFile https://github.com/qgis/QGIS/blob/master/src/core/qgsproject.cpp#L1120 https://github.com/qgis/QGIS/blob/master/src/core/qgsproject.cpp#L1507
* emission https://github.com/qgis/QGIS/blob/master/src/core/qgsproject.cpp#L1507
* connection https://github.com/qgis/QGIS/blob/master/src/core/qgsrelationmanager.cpp#L30
* readProject for relations https://github.com/qgis/QGIS/blob/master/src/core/qgsrelationmanager.cpp#L172

**Comment by rldhont** [comment]
@dmarteau can we made a test with py-qgis-server ?

**Comment by Gustry** [comment]
This has been fixed in QGIS 3.10.2 and master.
QGIS 3.4 should be used with cautious about relation. I will add a note.
